src/ansiblelint/rules/__init__.py

```diff
@@ -176,6 +179,16 @@ def matchtasks(self, file: Lintable) -> List[MatchError]:
             )
             match.task = task
             matches.append(match)
+
+        # If playbook contains blocks in pre_tasks/tasks/post_tasks
+        if self.needs_block and file.kind == "playbook":
```

Comment:
Do we really need the needs_block? IMHO, i would prefer to see matchtasks to have a recursive behavior naturally, calling matchpath for each found deep task.

Am I missing something there? Would enabling this break other existing rules?

(Just a question, not a request to change anything)

Reply:
Thanks for the comment.
I added it with reference to `needs_raw_task` for applying only to the specific rule.
Actually I have not tried it for other rules. I just did it and got some errors on pytest. If it is necessary to remove needs_block, then I will dig into and fix.

Reply:
I would rather not restrict this to only `file.kind == "playbook"` as well. Tasks files can also have nested blocks.

Reply:
I see. Got it. I'll then try to remove the the `if` line, and pass the tests.

Reply:
Hi, I added some commits and removed this statement. So let me close this review.